packages/mui-material/src/CircularProgress/CircularProgress.js

```diff
@@ -39,13 +39,22 @@ const circularDashKeyframe = keyframes`
   }
 `;
 
-const rotateAnimation = css`
-  animation: ${circularRotateKeyframe} 1.4s linear infinite;
-`;
+// This implementation is for supporting both Styled-components v4+ and Pigment CSS.
+// A global animation has to be created here for Styled-components v4+ (https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#12).
+// which can be done by checking typeof indeterminate1Keyframe !== 'string' (at runtime, Pigment CSS transform keyframes`` to a string).
+const rotateAnimation =
+  typeof circularRotateKeyframe !== 'string'
+    ? css`
+        animation: ${circularRotateKeyframe} 1.4s linear infinite;
+      `
+    : null;
```

Comment:
I decided to move the logic here so that `rotateAnimation` is not generated for Pigment CSS.